### ISSUE TITLE
T-SQL: Multiple BEGIN/END blocks in a file fail to parse correctly

### ISSUE THREAD

**Issue by fank** [quarylabs/sqruff#1809]
## Problem Description

When a T-SQL file contains multiple BEGIN/END blocks, only the first block is parsed correctly. Subsequent blocks are parsed as unparsable raw tokens, creating multiple file nodes in the parse tree.

## SQL that fails to parse

```sql
BEGIN
    SELECT 'Weekend';
END

BEGIN;
    SELECT 'Weekend';
END;

BEGIN;
    SELECT 'Weekend';
END

BEGIN
    SELECT 'Weekend';
END;
```

## Current behavior

The parser creates:
1. A properly parsed statement for the first BEGIN/END block
2. A second file node containing unparsable raw tokens for everything after the first block

This results in YAML output with multiple `- file:` entries, which is detected as an error by the CI check script.

## Expected behavior

All BEGIN/END blocks should be parsed correctly as separate statements within a single file node.

## Analysis

The issue appears to be related to:
1. How the parser handles multiple top-level statements in a file
2. Possible confusion between BEGIN/END blocks and BEGIN TRANSACTION statements when semicolon follows BEGIN
3. The test framework creating multiple file nodes when parsing fails

## Additional context

- This issue was discovered when adding the SQLFluff test case `test/fixtures/dialects/tsql/begin_end.sql`
- The CI check script (`.hacking/scripts/check_for_unparsable.sh`) correctly identifies this as a parsing failure
- Both `BEGIN` and `BEGIN;` syntaxes should be valid in T-SQL

## Related code

- BeginEndBlockSegment definition in `crates/lib-dialects/src/tsql.rs`
- FileSegment handling for T-SQL
- Statement parsing and delimiter handling

**Comment by Fank** [comment]
## Additional Finding: Blank Lines Trigger Multiple File Nodes

While investigating this issue, I discovered that blank lines (especially double blank lines) between statements also trigger the creation of multiple file nodes. This affects several test files including:

- print.sql (from SQLFluff)
- begin_end.sql (from SQLFluff)  
- Existing files when regenerated with blank lines

### Example that triggers the issue:

```sql
DECLARE @TestVal VARCHAR(20) = 'Test Print'


PRINT '#Dates'

PRINT CAST(GETDATE() AS VARCHAR(50));

PRINT @TestVal
```

The double blank line after the DECLARE statement causes the parser to create a new file node for everything after it.

This suggests the issue is broader than just BEGIN/END blocks - it's related to how the T-SQL parser handles statement separation and file structure when encountering blank lines.
